src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -6,227 +6,199 @@
+      val tsConverted = tsList.map(configConvert.value.to)
+      ConfigValueFactory.fromIterable(tsConverted.toIterable.asJava)
     }
   }
 
-  /**
-   * Extract the key from a fullKey, i.e. a namespace followed by a key. This function
-   * returns an error if the key contains the [[namespaceSep]], else the key found.
-   *
-   * {{{
-   *   scala> getMapKeyFrom("foo.bar", "foo")
-   *   res0: Try[String] = Success("bar")
-   *   scala> getMapKeyFrom("foo.bar.a", "foo")
-   *   res1: Try[String] = Failure(..)
-   * }}}
-   *
-   * @param fullKey The namespace plus key from which we want to extract the key
-   * @param namespace The namespace
-   * @return A [[Success]] with the key if it is possible to extract it, else a `Failure` with
-   *         the error
-   */
-  private[this] def getMapKeyFrom(fullKey: String, namespace: String): Try[String] = {
-    val substrStart = namespace.length + (if (namespace.isEmpty) 0 else namespaceSep.length)
-    val key = fullKey.substring(substrStart)
-    if (key contains namespaceSep) {
-      Failure(new RuntimeException(s"Invalid Map key found '$key'. Maps keys cannot contain the namespace separator '$namespaceSep'"))
-    } else {
-      Success(key)
-    }
-  }
-
-  implicit def deriveMap[T](implicit stringConvert: Lazy[StringConvert[T]]) = new ConfigConvert[Map[String, T]] {
-
-    override def from(config: RawConfig, namespace: String): Try[Map[String, T]] = {
-      val keysFound = (if (namespace.isEmpty)
-        config.keys
-      else
-        config.keys.filter(_ startsWith (namespace + namespaceSep))).toList
-
-      keysFound.foldLeft(Try(Map.empty[String, T])) {
-        case (f @ Failure(_), _) => f
-        case (Success(acc), fullKey) =>
-          for {
-            key <- getMapKeyFrom(fullKey, namespace)
-            rawValue <- Try(config(fullKey))
-            value <- stringConvert.value.from(rawValue)
-          } yield acc + (key -> value)
+  implicit def deriveMap[T](implicit configConvert: Lazy[ConfigConvert[T]]) = new ConfigConvert[Map[String, T]] {
+
+    override def from(config: ConfigValue): Try[Map[String, T]] = {
+      config match {
+        case co: ConfigObject =>
+          val keysFound = co.keySet().asScala.toList
+
+          keysFound.foldLeft(Try(Map.empty[String, T])) {
+            case (f @ Failure(_), _) => f
+            case (Success(acc), key) =>
+              for {
+                rawValue <- Try(co.get(key))
+                value <- configConvert.value.from(rawValue)
+              } yield acc + (key -> value)
+          }
+        case other =>
+          Failure(new Exception(s"Couldn't derive map from $other."))
       }
     }
 
-    override def to(keyVals: Map[String, T], namespace: String): RawConfig = {
-      keyVals.map { case (key, value) => makeNamespace(namespace, key) -> stringConvert.value.to(value) }
+    override def to(keyVals: Map[String, T]): ConfigValue = {
+      ConfigValueFactory.fromMap(keyVals.asJava)
```

Comment:
I suggest to use the `configConvert` for `T` here instead of delegating the conversion to typesafe config because that kind of conversion is what `pureconfig` exists for. Use something like:

```scala
ConfigValueFactory.fromMap(keyValues.mapValues(configConvert.value.to).asJava)
```